test_package/CMakeLists.txt

```diff
@@ -0,0 +1,15 @@
+project(type_safe_test_package)
+cmake_minimum_required(VERSION 2.8)
+
+include(${CMAKE_BINARY_DIR}/conanbuildinfo.cmake)
+include(${CONAN_CMAKE-UTILS_ROOT}/conan.cmake)
```

Comment:
This extra include of conan.cmake might be unnecessary. You might want to update to modern CMake syntax with ``conan_basic_setup(TARGETS)``